ion-hash/src/lib.rs

```diff
@@ -63,31 +83,31 @@ impl Markers {
 /// [`Digest`] algorithm.
 pub struct IonHasher<D>
 where
-    D: Digest,
+    D: IonHashDigest,
 {
     hasher: D,
 }
 
 impl<D> IonHasher<D>
 where
-    D: Digest,
+    D: IonHashDigest,
 {
-    fn new(hasher: D) -> Self {
+    pub fn new(hasher: D) -> Self {
         Self { hasher }
     }
 
     /// Computes the Ion hash over an [`Element`] recursively using this
     /// hasher's [`Digest`]
-    fn hash_element<E: Element + ?Sized>(&mut self, elem: &E) -> IonResult<Output<D>> {
+    pub fn hash_element<E: Element + ?Sized>(mut self, elem: &E) -> IonResult<Vec<u8>> {
```

Comment:
The change here (from `sha2::Digest::Output` to `Vec<u8>`) is temporary. I just don't really care about the abstraction right now. Make it work.. make it fast.